### ISSUE TITLE
The data with set 2 cannot be reproduced.

### ISSUE THREAD

**Issue by yuqiushi55** [taozh2017/UPRENet#1]
Dear author: In the case of setting 2, how should we set the TEST_LABEL and EXCLUDE_LABEL for the training script and test script? We set TEST_LABEL=EXCLUDE_LABEL=[2,3] and TEST_LABEL=EXCLUDE_LABEL=[1,4] for grouped validation, but the dice score is far from the data set in setting 2 in the paper. However, the data set in setting 1 can be reproduced. Here is the log under the setting of TEST_LABEL=EXCLUDE_LABEL=[2,3]: INFO - main - Test Class: RK
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_1.nii.gz.
INFO - main - Dice score: 0.6525903940200806
INFO - main - 1 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_2.nii.gz.
INFO - main - Dice score: 0.5593993067741394
INFO - main - 2 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_5.nii.gz.
INFO - main - Dice score: 0.5745718479156494
INFO - main - 5 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_8.nii.gz.
INFO - main - Dice score: 0.5332025289535522
INFO - main - 8 has been saved. 
INFO - main - Test Class: RK
INFO - main - Mean class IoU: 0.40981030464172363
INFO - main - Mean class Dice: 0.5799410343170166
INFO - main - Test Class: LK
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_1.nii.gz.
INFO - main - Dice score: 0.6142765879631042
INFO - main - 1 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_2.nii.gz.
INFO - main - Dice score: 0.61604905128479
INFO - main - 2 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_5.nii.gz.
INFO - main - Dice score: 0.7876827716827393
INFO - main - 5 has been saved. 
INFO - main - Tested query volume: /chaos_MR_T2_normalized/image_8.nii.gz.
INFO - main - Dice score: 0.5100481510162354
INFO - main - 8 has been saved. 
INFO - main - Test Class: LK
INFO - main - Mean class IoU: 0.47012144327163696
INFO - main - Mean class Dice: 0.6320141553878784
INFO - main - Final results...
INFO - main - Mean IoU: {'RK': 0.40981030464172363, 'LK': 0.47012144327163696}
INFO - main - Mean Dice: {'RK': 0.5799410343170166, 'LK': 0.6320141553878784}
INFO - main - Total Mean Dice: 0.6059775948524475
INFO - main - End of validation.
INFO - UPRENet - Result: 1
INFO - UPRENet - Completed after 0:00:31

**Comment by Jeff-Hu0401** [comment]
You may try training with pseudo-labels while excluding all images that contain the target test organs during training, i.e., setting EXCLUDE_LABEL = {1, 2, 3, 4}.

**Comment by LW-J-King** [comment]
Hi yuqiushi55,
I saw your issue about reproducing Setting 2 on the dataset, where you set TEST_LABEL=EXCLUDE_LABEL=[2,3] and [1,4] but got much lower Dice than the paper. I’m currently facing the exact same problem on Synapse-CT/SABS when replicating Setting 2.
May I ask a few follow-up questions?
1.Have you successfully reproduced the official full Dice results of Setting 2 as shown in the paper after adjusting your configuration?
2.When you set EXCLUDE_LABEL, did you apply the same exclude list for both upper and lower organ subgroups?
3.How did you implement the slice filtering logic in dataloaders/datasets.py? Did you remove slices that contain any label inside exclude_label (OR logic), or only slices with all excluded labels at once?
4.Besides modifying TEST_LABEL and EXCLUDE_LABEL, did you adjust other hyperparameters like use_gt, n_part, supp_idx, or modify the train-test data split for Setting 2?
5.Could you share the full training and testing command you finally used if you managed to match the paper’s metrics?
It would be really helpful to exchange reproduction experience with you, thanks a lot!

**Comment by yuqiushi55** [comment]
1. After adjusting the configuration, the results in the paper can be basically reproduced
2. In the case of setting 2, EXCLUDE_LABEL should remain consistent with TEST_LABEL
3. Don't worry about the logic in datasets.py, just set EXCLUDE_LABEL in the training script
4. No need to change use_gt, n_part, or supp_idx
5. The time since the reproduction is a bit long, and there is no record of the reproduction data. The reproduction results are close to those in the paper, with little difference
In summary, for setting 2, it is sufficient to set EXCLUDE_LABEL to be consistent with TEST_LABEL in the training script
